### ISSUE TITLE
Typescript: Incomplete definition for ClientOptions.node

### ISSUE THREAD

**Issue by mauriciovillalobos** [elastic/elasticsearch-js#881]
## 🐛 Bug Report

The typescript definition for `ClientOptions` describe the `node` field as:
```ts
interface ClientOptions {
  node?: string | string[];
  nodes?: string | string[];
  ...
}
```

However the docs show a possible way of passing an object instead:
```js
node: {
  url: new URL('http://localhost:9200'),
  ssl: 'ssl options',
  agent: 'http agent options',
  id: 'custom node id',
  headers: { 'custom': 'headers' }
  roles: {
    master: true,
    data: true,
    ingest: true,
    ml: false
  }
}
```
source https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/client-configuration.html#_basic_options

Trying to do so results in the obvious typescript error
```
Object literal may only specify known properties, and 'url' does not exist in type 'string[]'.
```

## Expected behavior

The `ClientOptions` Interface should include a thorough typing for the allowed values on the `node` field

## Your Environment

- *node version*: 8.12.0
- `@elastic/elasticsearch` *version*: >=7.1.0
- *os*: Mac
- Typescript *version*: 3.5.1

**Comment by delvedor** [comment]
Hello! You are right, thanks for reporting!

**Comment by delvedor** [comment]
Closed in #882.
